tun/src/tokio/mod.rs

```diff
@@ -15,6 +15,13 @@ impl TunInterface {
         Ok(Self { inner: AsyncFd::new(tun)? })
     }
 
+    #[instrument]
+    pub async fn set_up(&self, up: bool) -> io::Result<()> {
+        let mut guard = self.inner.readable().await?;
+        guard.try_io(|inner| inner.get_ref().set_up(up));
+        Ok(())
```

Comment:
This should fix the hang:
```suggestion
        self.inner.get_ref().set_up(up)
```

You were waiting for `self` to become readable, which means waiting for a network packet